code/modules/antagonists/heretic/heretic_knowledge.dm

```diff
@@ -0,0 +1,420 @@
+
+/**
+ * # Heretic Knowledge
+ *
+ * The datums that allow heretics to progress and learn new spells and rituals.
+ *
+ * Heretic Knowledge datums are not singletons - they are instantiated as they
+ * are given to heretics, and deleted if the heretic antagonist is removed.
+ *
+ */
+/datum/heretic_knowledge
```

Comment:
i know this will be a lot of work but i heavily suggest you port on of yog's prs that decoupled knowledge from actual rituals. that'd allow you to bundle up all the curses into one knowledge and put something more... creative in the slot that's left empty. https://github.com/yogstation13/Yogstation/pull/11003